packages/design-system/src/components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,184 @@
+
+    this.setState({
+      searchTerm: value
+    });
+
+    if (debounceInMs) {
+      this.debouncedOnChange(value);
+    } else {
+      onChange(value);
+    }
+  };
+
+  handleKeyPress = key => {
+    const { onSubmit } = this.props;
+
+    if (onSubmit) {
+      if (key.keyCode === KeyCodes.enter) {
+        onSubmit(this.state.searchTerm);
+      }
+    }
+  };
+
+  handleClear = () => {
+    const { onChange, onSubmit } = this.props;
+    const searchFunction = onSubmit || onChange;
+
+    this.setState({
+      searchTerm: ''
+    });
+
+    // 1) Cancel debounced search
+    // 2) Use onClear instead of onSubmit | onChange function
+    searchFunction('');
+
+    if (this.inputRef.current) {
+      this.inputRef.current.focus();
+    }
+  };
+
+  handleCloseIconKeyDown = key => {
+    if (key.keyCode === KeyCodes.enter) {
+      this.handleClear();
+    }
+  };
+
+  toggleCompactMode = () => {
+    this.setState(
+      prevState => ({
+        isInCompactMode: !prevState.isInCompactMode
+      }),
+      () => {
+        if (!this.state.isInCompactMode && this.inputRef.current) {
+          this.inputRef.current.focus();
+        }
+      }
+    );
+  };
+
+  render() {
+    const { className, placeholder, value, loading, size, error } = this.props;
+    const { searchTerm, isInCompactMode } = this.state;
+
+    const isCompactSize = size === 'compact';
+    const shouldDisplayCloseButton = searchTerm && !loading && !isInCompactMode;
+
+    return (
+      <span className={className}>
+        <div className={styles[`${baseClass}__container`]}>
+          <SearchIcon
+            fill="#424d57"
+            width="18px"
+            height="18px"
+            className={cx(`lc-${baseClass}__icon--search`, {
+              [`lc-${baseClass}__icon--search-compact`]: isCompactSize
+            })}
+            onClick={isCompactSize ? this.toggleCompactMode : null}
+          />
+          {loading && (
+            <Loader
+              size="small"
+              className={styles[`${baseClass}__icon--loader`]}
+            />
+          )}
+          <Input
+            placeholder={placeholder}
+            ref={this.inputRef}
+            value={value || this.state.searchTerm}
+            onChange={this.handleChange}
+            onKeyDown={this.handleKeyPress}
+            className={cx(`lc-${baseClass}__input`, {
+              [`lc-${baseClass}__input-compact`]: isInCompactMode
+            })}
+          />
+          {shouldDisplayCloseButton && (
+            <CloseIcon
+              fill="#424d57"
+              width="18px"
+              height="18px"
+              onClick={this.handleClear}
+              onKeyDown={this.handleCloseIconKeyDown}
+              className={styles[`${baseClass}__icon--close`]}
+              tabIndex="0"
+            />
+          )}
+        </div>
+        {error && <FieldError>{error}</FieldError>}
+      </span>
+    );
+  }
+}
+
+const basePropTypes = {
```

Comment:
Why do you create a const if you can set it directly in `SearchBar`? It's not exported or something.

Reply:
You're correct. I've looked on many DS components and tried to do the same (to follow convention)